instrumentation/cassandra/cassandra-4.0/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_0/CassandraTelemetry.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.cassandra.v4_0;
+
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbClientSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.SqlClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetClientAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+
+public final class CassandraTelemetry {
+
+  public static CassandraTelemetryBuilder builder(OpenTelemetry openTelemetry) {
+    return new CassandraTelemetryBuilder(openTelemetry);
+  }
+
+  private final Instrumenter<CassandraRequest, ExecutionInfo> instrumenter;
+
+  CassandraTelemetry(
+      OpenTelemetry openTelemetry,
+      String instrumentationName,
+      boolean statementSanitizationEnabled) {
+    this.instrumenter =
+        createInstrumenter(openTelemetry, instrumentationName, statementSanitizationEnabled);
+  }
+
+  private static Instrumenter<CassandraRequest, ExecutionInfo> createInstrumenter(
+      OpenTelemetry openTelemetry, String instrumenterName, boolean statementSanitizationEnabled) {
+    CassandraSqlAttributesGetter attributesGetter = new CassandraSqlAttributesGetter();
+
+    return Instrumenter.<CassandraRequest, ExecutionInfo>builder(
+            openTelemetry, instrumenterName, DbClientSpanNameExtractor.create(attributesGetter))
+        .addAttributesExtractor(
+            SqlClientAttributesExtractor.builder(attributesGetter)
+                .setTableAttribute(SemanticAttributes.DB_CASSANDRA_TABLE)
+                .setStatementSanitizationEnabled(statementSanitizationEnabled)
+                .build())
+        .addAttributesExtractor(
+            NetClientAttributesExtractor.create(new CassandraNetAttributesGetter()))
+        .addAttributesExtractor(new CassandraAttributesExtractor())
+        .buildInstrumenter(SpanKindExtractor.alwaysClient());
+  }
+
+  public Instrumenter<CassandraRequest, ExecutionInfo> getInstrumenter() {
+    return instrumenter;
+  }
```

Comment:
I think I didn't do that because how can we use the `TracingCqlSession` with the reactive method'? In this way the `CassandraTelemetry` (shared in both javaagent module: cassandra4.0 and cassandra4.4) will decorate only the original TracingCqlSession.

Reply:
Maybe this is not needed if I simply wrap the original tracingCql. I did a [clean-up](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6441/commits/39b3d728cd6051d56c6796b05e532f988f675190).

Reply:
[Done](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6441/commits/1895bfca1ae5ebc26ca091aab7bbad9032b6fdb4)!